lib/tree.rb

```diff
@@ -30,6 +30,14 @@ def contents_array
       @contents_ary ||= jtree_entries
     end
     
+    def recursive_count(limit = nil)
+      jtree_entries({recursive: true, limit: limit}).size
```

Comment:
Would it make sense to cache the results of this operation? Or not, because of the possible different `limits`?